packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -295,14 +295,25 @@ class DBot {
      * Checks whether the workspace contains all required blocks before running the strategy.
      */
     checkForRequiredBlocks() {
+        let error;
+
         if (!hasAllRequiredBlocks(this.workspace)) {
-            const error = new Error(
+            error = new Error(
                 localize(
                     'One or more mandatory blocks are missing from your workspace. Please add the required block(s) and then try again.'
                 )
             );
-            globalObserver.emit('Error', error);
+        }
+        if (!isAllRequiredBlocksEnabled(this.workspace)) {
```

Comment:
Is it possible to merge these 2 if statements into one? their fulfillment statements are the same.